src/app/requests.ts

```diff
@@ -0,0 +1,55 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
```

Comment:
It seems to me the bulk of the change in `page.tsx` is to bring in the use of `react-query` to simplify things. I haven't seen/used this library before, but it seems pretty neat :tada: .